pkg/services/notification/handler.go

```diff
@@ -75,83 +79,208 @@ func (s *Server) CreateNotification(ctx context.Context, req *pb.CreateNotificat
 		if err != nil {
-			logger.Errorf(ctx, "Push task [%s] into etcd failed, [%+v]", task.TaskId, err)
+			logger.Errorf(ctx, "Push task[%s] into etcd failed, [%+v].", task.TaskId, err)
 			return nil, err
 		}
+		logger.Debugf(ctx, "Push task[%s] into etcd successfully.", task.TaskId)
+		logger.Debugf(ctx, "Create Task[%s] successfully.", task.TaskId)
 	}
 
-	// Enqueue notification after tasks.
-	err = s.controller.notificationQueue.Enqueue(notification.NotificationId)
-	if err != nil {
-		logger.Errorf(ctx, "Push notification [%s] into etcd failed, [%+v]", notification.NotificationId, err)
-		return nil, err
-	}
-
-	return &pb.CreateNotificationResponse{
-		NotificationId: pbutil.ToProtoString(notification.NotificationId),
-	}, nil
+	return tasks, nil
 }
 
 func (s *Server) DescribeNotifications(ctx context.Context, req *pb.DescribeNotificationsRequest) (*pb.DescribeNotificationsResponse, error) {
 	nfs, nfCnt, err := rs.DescribeNotifications(ctx, req)
 	if err != nil {
-		logger.Errorf(ctx, "Failed to describe notifications, error: %+v", err)
+		logger.Errorf(ctx, "Failed to describe notifications, [%+v].", err)
 		return nil, gerr.NewWithDetail(ctx, gerr.Internal, err, gerr.ErrorDescribeResourcesFailed)
 	}
 	nfPbSet := models.ParseNfSet2PbSet(nfs)
 	res := &pb.DescribeNotificationsResponse{
 		TotalCount:      uint32(nfCnt),
 		NotificationSet: nfPbSet,
 	}
-	return res, nil
-}
 
-func (s *Server) RetryNotifications(ctx context.Context, req *pb.RetryNotificationsRequest) (*pb.RetryNotificationsResponse, error) {
-	return &pb.RetryNotificationsResponse{}, nil
+	logger.Debugf(ctx, "Describe notifications successfully, notifications=[%+v].", res)
+	return res, nil
 }
 
 func (s *Server) DescribeTasks(ctx context.Context, req *pb.DescribeTasksRequest) (*pb.DescribeTasksResponse, error) {
 	tasks, taskCnt, err := rs.DescribeTasks(ctx, req)
 	if err != nil {
-		logger.Errorf(ctx, "Failed to describe tasks, error: %+v", err)
+		logger.Errorf(ctx, "Failed to describe tasks,[%+v].", err)
 		return nil, gerr.NewWithDetail(ctx, gerr.Internal, err, gerr.ErrorDescribeResourcesFailed)
 	}
 	taskPbSet := models.ParseTaskSet2PbSet(tasks)
 	res := &pb.DescribeTasksResponse{
 		TotalCount: uint32(taskCnt),
 		TaskSet:    taskPbSet,
 	}
+	logger.Debugf(ctx, "Describe tasks successfully, tasks=[%+v].", res)
+	return res, nil
+}
+
+func (s *Server) RetryNotifications(ctx context.Context, req *pb.RetryNotificationsRequest) (*pb.RetryNotificationsResponse, error) {
+	nfIds := stringutil.SimplifyStringList(req.NotificationId)
+	nfs, err := rs.GetNfsByNfIds(req.NotificationId)
+	if err != nil {
+		logger.Errorf(ctx, "Failed to retry notifications[%+v], [%+v].", nfIds, err)
+		return nil, gerr.NewWithDetail(ctx, gerr.Internal, err, gerr.ErrorRetryTaskFailed, nfIds)
+	}
+	if len(nfs) == 0 {
+		logger.Infof(ctx, "Retry notifications[%+v] do not exit.", nfIds)
+		return nil, gerr.NewWithDetail(ctx, gerr.Internal, err, gerr.ErrorRetryNotificationtNotExist, nfIds)
```

Comment:
Not found the notification, change error from Internal to NotFound.

Reply:
updated.